mllib/src/test/scala/org/apache/spark/ml/clustering/GaussianMixtureSuite.scala

```diff
@@ -126,9 +143,93 @@ class GaussianMixtureSuite extends SparkFunSuite with MLlibTestSparkContext
     testEstimatorAndModelReadWrite(gm, dataset,
       GaussianMixtureSuite.allParamSettings, checkModelData)
   }
+
+  test("univariate dense/sparse data with two clusters") {
+    val weights = Array(2.0 / 3.0, 1.0 / 3.0)
+    val means = Array(Vectors.dense(5.1604), Vectors.dense(-4.3673))
+    val covs = Array(Matrices.dense(1, 1, Array(0.86644)), Matrices.dense(1, 1, Array(1.1098)))
+    val gaussians = means.zip(covs).map { case (mean, cov) =>
+      new MultivariateGaussian(mean, cov)
+    }
+    val expected = new GaussianMixtureModel("dummy", weights, gaussians)
+
+    Seq(denseDataset, sparseDataset).foreach { dataset =>
+      val actual = new GaussianMixture().setK(2).setSeed(seed).fit(dataset)
+      modelEquals(expected, actual)
+    }
+  }
+
+  test("check distributed decomposition") {
```

Comment:
Yeah, I also suffer from bad initialization in some of my use cases. So I think we should push to commit [SPARK-15785](https://issues.apache.org/jira/browse/SPARK-15785) firstly. It's more easy to add correctness test after we support initial model. I'll leave this as follow up and open [SPARK-19144](https://issues.apache.org/jira/browse/SPARK-19144) to track. Thanks.